### ISSUE TITLE
The chronology part is misleading

### ISSUE THREAD

**Issue by ayaka14732** [FranxYao/Deep-Generative-Models-for-Natural-Language-Processing#1]
Consider the two definitions of DGM4NLP:

(1) The model can generate texts (Transformer, GPT3, ...)
(2) In addition to (1), given the same input, the model can generate a set of different texts (VAE, GAN, ...)

BERT is certainly not a typical DGM because it only keeps the encoder part of the Transformer architecture. It is more commonly considered a discriminative model rather than a generative model.

Transformer and GPT3 do not meet definition (2) because they do not contain a latent variable to control the generation process.

**Comment by ayaka14732** [comment]
As far as I know, the most qualified model that can be considered as a DGM4NLP is [transformer-vae](https://github.com/Fraser-Greenlee/transformer-vae), but the result is not satisfactory either.

**Comment by FranxYao** [comment]
Hi @ayaka14732 

Thank you for asking. Actually, when I am compiling the chronology part, I was more about including important works in either one of probabilistic modeling or NLP, but not necessarily right in the intersection. 

Yet it would also be great to formally discuss the naming and definition. To me, instead of strictly differentiating discriminative models and generative models, I usually think of a graphical model (typically undirected) where there are certain observed variables and certain latent variables. The real problem is how to do inference of a given model.

Translating into NLP tasks, think about the relationship between supervised constituency parsing, unsupervised grammar induction, and partially supervised constituency parsing (where certain spans in a sentence are labeled as a constituent but not all spans). 
* In supervised constituency parsing (all constituents are observed), one usually uses a hypertree structured conditional random field, which is a typical discriminative model p(t | x) where t is the tree and x is the sentence. See: https://www.ijcai.org/proceedings/2020/0560.pdf
* In unsupervised constituency grammar induction (no constituents are observed), one usually uses a probabilistic context free grammar, which is a typical generative model p(t, x), t being the tree and x being the sentence. See: https://arxiv.org/abs/2104.13727
* But in the case where part of the constituents (i.e., some parts in t) are observed but not all, what shall we call the model? Discriminative or generative? Well, in such cases it would be difficult to differentiate because part of t is observed and part of t is latent. see  https://arxiv.org/abs/2012.08478

The above example is usually the case when there exists a complicated generative process: certain factors are observed, certain factors are latent, certain factors, even observed, can be noisy, so we should probably do not trust the labels that much. The good thing with undirected graphical models is that they allow us to model all different variables in a unified and principled way. Yet when the underlying graph is complicated, it would be hard to claim the model is discriminative or generative.

**Comment by FranxYao** [comment]
@ayaka14732 

Continuing the above arguments, I would simply suggest just thinking of graphical models and forgo the classification whether they are discriminative or generative. If this is OK we could continue to discuss the relationship between VAEs and transformers. Here I would suggest differentiating the graphical / probabilistic model itself and the **parameterization** of the model:
* A probabilistic model is a probability distribution
* The parameterization of a probabilistic model is anything that produces the parameters of the distribution, which can be neural networks. In such cases, the community usually calls it neural parameterization. 

 For example, in VAEs we have an inference model and a generative model. 
* The inference model is a probability distribution: p(z | x) while x being the observed and z being the latent 
* The parameterization of the inference model can be a neural network, and specifically, it can be a pretrained BERT. In such cases one can say "we use a pretrained BERT as the neural parameterization of our inference model". 
* Similarly, the generative model is a joint distribution p(x, z)
* The parameterization of the generative model can also be a neural network. 

As an example, in OPTIMUS https://arxiv.org/abs/2004.04092, the inference model is parameterized as a pretrained BERT and the generative model is parameterized as a pretrained GPT. 

I hope this clarifies the relationship between BERT and VAEs. BERT is not VAE definitely, but it can be part of the VAE, i.e., the inference model can be parameterized as BERT. In fact, nowadays I believe every inference model in this area should at least be parameterized with a BERT base. 

For performance improvements, the quantized transformer space is a good example: https://arxiv.org/pdf/2012.04443.pdf It usually transformers as the parameterization for the encoder and decoder of a VQVAE and performs well on opinion summarization.

I hope this clarifies the questions

**Comment by ayaka14732** [comment]
I think I get what you mean. Previously I thought you were saying that VAE/GAN/BERT/GPT3 are all deep generative models, but instead you meant VAE/GAN are generative models and BERT/GPT3 are deep models.

**Comment by FranxYao** [comment]
Exactly. Generative models are nice models where we can perform all types of inference. Deep models are nice models that are just powerful. The combination of them is where the cool and fun things happen.
